Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -1472,10 +1472,15 @@ ThrowCompletionOr<bool> is_loosely_equal(GlobalObject& global_object, Value lhs,
         // b. If ℝ(x) = ℝ(y), return true; otherwise return false.
         if ((lhs.is_number() && !lhs.is_integral_number()) || (rhs.is_number() && !rhs.is_integral_number()))
             return false;
-        if (lhs.is_number())
-            return Crypto::SignedBigInteger { MUST(lhs.to_i32(global_object)) } == rhs.as_bigint().big_integer();
-        else
-            return Crypto::SignedBigInteger { MUST(rhs.to_i32(global_object)) } == lhs.as_bigint().big_integer();
+
+        VERIFY(!lhs.is_nan() && !rhs.is_nan());
+        if (lhs.is_number()) {
+            return rhs.as_bigint().big_integer().compare_to_double(lhs.as_double())
+                == Crypto::SignedBigInteger::CompareResult::DoubleEqualsBigInt;
+        } else {
+            return lhs.as_bigint().big_integer().compare_to_double(rhs.as_double())
+                == Crypto::SignedBigInteger::CompareResult::DoubleEqualsBigInt;
+        }
```

Comment:
what's even happening here

Reply:
One side is a big int the other a number (double or i32).
I also dislike this way of doing it but this is how it was done previously.
I can change this equals part but the less than part get confusing because you have to swap the result so I'll keep that one as is.

(Unless you're talking about the line splitting which is because the enum name is so long)

Reply:
nah, the line splitting is weird, but I get the rationale, the way it's written was confusing - your new edit is much nicer.